target/linux/ramips/base-files/etc/hotplug.d/firmware/10-rt2x00-eeprom

```diff
@@ -46,6 +46,11 @@ case "$FIRMWARE" in
 		rt2x00_eeprom_extract "factory" 0 512
 		rt2x00_eeprom_set_macaddr $wifi_mac
 		;;
+	apac1200)
+		wifi_mac=$(mtd_get_mac_ascii config INIC_MAC_ADDR)
```

Comment:
Something seem to me wrong. You are reading the mac address from the very same partition as binary in case of the ethernet mac address. But here you need to do an ascii read. Does the partition have ascii content or binary content?

Does the factory partition includes the mac address by any chance? would you please attach a dump of the factory partition.